### ISSUE TITLE
Synchronize does not use playbook directory in 2.1

### ISSUE THREAD

**Issue by iquito** [ansible/ansible#14944]
##### Issue Type:

- Bug Report
##### Ansible Version:

```
ansible 2.1.0
```
##### Summary:

I have been using synchronize the following way (update-config.yml):

```
synchronize:
      src: config/
      dest: /directory/config
      delete: yes
      recursive: yes
```

and called the playbook with

```
ansible-playbook directory/update-config.yml
```

This worked fine in ansible 1.9.2, but in 2.1, it tries to resolve config/ from the directory I am in, not the directory the playbook is in, meaning it looks for `config/` instead of `directory/config/` and does not find anything (or in other scenarios it might just find the wrong stuff).

**Comment by tima** [comment]
Can you verify this is an issue in 2.0 here to pinpoint the issue? There's a lot of changes to this module and action plugin between 1.9.2 and devel, the in-progress work towards v2.1.

**Comment by iquito** [comment]
I did a `git clone git://github.com/ansible/ansible.git --recursive` 2 days ago, before I reported the issue, so it should be the absolutely newest version.

It should also be easy to reproduce. If you need any more information or if there is anything to test, I will happily try it out.

**Comment by abadger** [comment]
I believe this was an intentional change that happened globally because current working directory is insecure.  If you are in /var/tmp when you run the playbook, another user could have put a different config file into place.

@bcoca could you let me know if my supposition above is correct?  Or should synchronize be looking for those files in the current working directory?

**Comment by abadger** [comment]
bcoca pointed out that I misread the report.  It's now looking in the cwd instead of in the playbook basedir and so it's now wrong.  Taking a look.

**Comment by abadger** [comment]
Okay, try out pull request #15167 I just tested locally and it seems to do the right thing now.

**Comment by abadger** [comment]
Pull request merged to devel and cherry-picked to stable-2.0 branch.  This should be in 2.1.0 and either 2.0.2 (we've already started the release process for 2.0.2 so not sure if it will make it in there) or 2.0.3 if we make a release for that.

**Comment by iquito** [comment]
Just tested it - this is definitely fixed in devel. Thanks!
